### PR TITLE
[Issue 9480][pulsar-io] add option for auto.offset.reset to kafka source

### LINKED ISSUE #9480: support auto.offset.reset in kafka connector
**Is your enhancement request related to a problem? Please describe.**
The kafka source sets auto.offset.reset to "earliest". This means all old messages from kafka are produced to pulsar. Often is it desirable to start form the present location "latest".
The option is set after the user config has been loaded so it cannot be changed:
[source code link](https://github.com/apache/pulsar/blob/master/pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java#L87)

**Describe the solution you'd like**
Make this an option of the kafka source config.

**Describe alternatives you've considered**
Setting the option in the consumerConfigProperties of the source. This is not possible because the source sets the option after the user config has been loaded.

**Additional context**

### PR DESCRIPTION
**Pull request by sbourkeostk**
Fixes #9480 

### Motivation

The kafka source sets auto.offset.reset to "earliest". This means all old messages from kafka are produced to pulsar. Often is it desirable to start form the present location "latest".
The option is set after the user config has been loaded so it cannot be changed:
[source code link](https://github.com/apache/pulsar/blob/master/pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java#L87)

### Modifications

Added an autoOffsetReset option to KafkaSourceConfig

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change added tests and can be verified as follows:
Added to unit test and verified behaviour.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: no
  - The schema: no
  - The default values of configurations: no
  - The wire protocol: no
  - The rest endpoints: no
  - The admin cli options: no
  - Anything that affects deployment: no

### Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented? docs / JavaDocs
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: site2/docs/adaptors-kafka.md [hunk 1]
```diff
@@ -128,7 +128,7 @@ Properties:
 | Config property                         | Supported | Notes                                                                         |
 |:----------------------------------------|:----------|:------------------------------------------------------------------------------|
 | `acks`                                  | Ignored   | Durability and quorum writes are configured at the namespace level            |
-| `auto.offset.reset`                     | Yes       | Will have a default value of `latest` if user does not give specific setting. |
```
Comment (sbourkeostk) [inline site2/docs/adaptors-kafka.md:131]: This is a correction to an error in the existing docs.

**Comment by Renkai** [comment]
@BewareMyPower Could you have a look?

**Review by codelipenghui — APPROVED** [review]

**Review by BewareMyPower — CHANGES_REQUESTED** [review]
Could you add the sample config to `kafkaSourceConfig.yaml` and verify it in `loadFromYamlFileTest`?

**Comment by sbourkeostk** [comment]
Sample config added to kafkaSourceConfig.yaml and verified in loadFromYamlFileTest

**Review by BewareMyPower — APPROVED** [review]

File: site2/docs/adaptors-kafka.md [hunk 2]
```diff
@@ -128,7 +128,7 @@ Properties:
...
+| `auto.offset.reset`                     | Yes       | Will have a default value of `earliest` if user does not give specific setting. |
```
Comment (Anonymitaet) [inline site2/docs/adaptors-kafka.md:131]:
```suggestion
| `auto.offset.reset`                     | Yes       | It uses a default value of `earliest` if you do not give a specific setting. |
```
In technical information, follow these general guidelines for personal pronouns:
• Avoid the first-person pronouns I and we, except in these situations:
° In the question portion of frequently asked questions (FAQs)
° In articles, white papers, or documents that have listed authors and in which the authors describe their own actions or opinions
• Use the second-person pronoun you as much as possible. The subject of an imperative sentence is understood to be you.
• Avoid third-person pronouns that are gender specific.

**Review by sijie — APPROVED** [review]

**Review by Anonymitaet — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]
LGTM

**Review by 315157973 — APPROVED** [review]

**Review by jiazhai — APPROVED** [review]

**Pull request merged**